internal/pkg/cli/task_run.go

```diff
@@ -260,10 +260,11 @@ func (o *runTaskOpts) configureRunner() (taskRunner, error) {
 
 			OS: o.os,
 
-			VPCGetter:            vpcGetter,
-			ClusterGetter:        ecs.New(o.sess),
-			Starter:              ecsService,
-			EnvironmentDescriber: d,
+			VPCGetter:             vpcGetter,
+			ClusterGetter:         ecs.New(o.sess),
+			Starter:               ecsService,
+			EnvironmentDescriber:  d,
+			NonZeroExitCodeGetter: ecs.New(o.sess),
```

Comment:
We can reuse the same client for both `ClusterGetter` and `NonZeroExitCodeChecker`:

```suggestion
   ecsClient := ecs.New(o.sess)
   ...
			ClusterGetter:    ecsClient,
			Starter:               ecsService,
			EnvironmentDescriber:  d,
			NonZeroExitCodeGetter:ecsClient,
```